hydra/core/override_parser/types.py

```diff
@@ -26,6 +26,16 @@ class QuotedString:
 
     quote: Quote
 
+    def __hash__(self) -> int:
+        return hash(self.text)
+
+    def __eq__(self, other: Any) -> Any:
+        # We do not care whether quotes match for equality.
+        if isinstance(other, QuotedString):
+            return self.text == other.text
+        else:
+            return NotImplemented
+
```

Comment:
There are multiple tests that are comparing to QuotedString. This makes them softer.
What is the motivation for this change?

Reply:
So, my original motivation was that I had to implement `__hash__()` and this made me wonder what it meant for two quoted strings to be equal, and I thought it would make more sense to ignore quotes (just like `'xyz' == "xyz"` in Python). Since these quoted strings can now be dictionary keys, I also thought that if we ever wanted to merge two dictionaries, we may otherwise accidentally end up with a dictionary with two quoted strings representing the same string, but with different quotes... that being said, I have no idea if it's something that may happen in the current codebase.

I can definitely revert that change if you prefer.

Reply:
The quoted string objects are mostly used to reconstruct the quotes in some scenarios related to sweeping.
the objects that are merged to the config should be the string without the quotes.

For testing purposes, I do want to differentiate between the two types of quotes.

I think it's best to revert it.

Reply:
> I think it's best to revert it.

Done in e249eb5ff. Note that I made `QuotedString` immutable since the dataclass doc suggests it is safer for hashable types (see `unsafe_hash` in https://docs.python.org/3/library/dataclasses.html).

Reply:
that's fine because I don't actually think we are changing it after construction.